### ISSUE TITLE
v9 <Transition> bug with NextJS

### ISSUE THREAD

**Issue by aleclarson** [react-spring/react-spring#699]
https://github.com/react-spring/react-spring/issues/325

@aleclarson 
I try v9-beta with nextjs look like animation flashing again when I change route with animation.
It works fine on v8 and feels smoother.

v8 example
https://react-spring-325-aszmjdhnnv.now.sh
![May-30-2562 08-36-35](https://user-images.githubusercontent.com/4003032/58602256-74fe6a80-82b6-11e9-98ee-674b8d773217.gif)

v9 example
https://react-spring-325-tkuptdounw.now.sh
![May-30-2562 08-35-05](https://user-images.githubusercontent.com/4003032/58602131-db36bd80-82b5-11e9-8fce-321ebf2d14de.gif)

codesandbox (code example but this behavior does not effect on codesandbox)
https://codesandbox.io/s/reactspring-325-vu8en

try to change route between about page and contacts page slowly, you will see flashing white background before they animate.

_Originally posted by @rainstormza in https://github.com/react-spring/react-spring/issues/642#issuecomment-497167320_

**Comment by aleclarson** [comment]
@rainstormza Can you reproduce without NextJS?

**Comment by rainstormza** [comment]
@aleclarson I think it works when used with create-react-app
I think the problem is server-side-rendering.

@drcmda Does v8 fix about SSR problem?

**Comment by tim-soft** [comment]
@aleclarson HMR is broken when editing anything in `/pages` other than _app.js

updated sandbox with 9.0.0-beta.23
https://codesandbox.io/s/react-spring-325-rk3mo

**Comment by aleclarson** [comment]
@tim-soft Using the `reset` prop seems to fix it.

**Comment by aleclarson** [comment]
Also, the OP's issue was fixed somewhere between beta.8 and beta.27  🎉

**Comment by rainstormza** [comment]
@aleclarson Animation still flash on v9 beta.31
but it works fine on v8

**Comment by aleclarson** [comment]
Huh, that's weird...

**Comment by aleclarson** [comment]
This will definitely be fixed in the next canary version, but the example needs a small change to its `AnimatedRoute` component.

```diff 
 const AnimatedRoute = ({ children }) => (
   <Route
     render={({ location }) => (
       <Transition
         native
         items={location}
         keys={location => location.pathname}
         from={{
           opacity: 0,
           transform: 'perspective(900px) rotateY(180deg)',
         }}
         enter={[
           { opacity: 1, transform: 'perspective(1000px) rotateY(0deg)' },
           { transform: 'none', immediate: true },
         ]}
-        leave={{ opacity: 0, transform: 'perspective(900px) rotateY(-180deg)', pointerEvents: 'none' }}>
-        {location => style => <Container style={style}>{children(location)}</Container>}
+        leave={[
+          { transform: 'perspective(1000px) rotateY(0deg)', immediate: true },
+          {
+            opacity: 0,
+            transform: 'perspective(900px) rotateY(-180deg)',
+            pointerEvents: 'none',
+          },
+        ]}>
+        {(style, location) => (
+          <Container style={style}>{children(location)}</Container>
+        )}
       </Transition>
     )}
   />
 )
```
The first update of the `leave` animation helps avoid trying to animate from `none` to `perspective(900px) rotateY(-180deg)`, which is impossible because `none` doesn't contain 2 numeric strings like the other does.

Also, the render prop has a new signature now, in accordance with #809.

**Comment by aleclarson** [comment]
Whoops, I commented on the wrong issue. Sorry! This one will be fixed in the next canary too, though. 😅

**Comment by aleclarson** [comment]
Now available in v9.0.0-rc.2 #985

Demo here: https://codesandbox.io/s/react-spring-699-57yi5
